.hlint-universum.yaml

```diff
@@ -0,0 +1,414 @@
+# - warn: {lhs: fmap concat (for f s), rhs: Universum.concatForM s f}
+# - warn: {lhs: concat <$> forM f s, rhs: Universum.concatForM s f}
+# - warn: {lhs: concat <$> for f s, rhs: Universum.concatForM s f}
+
+- suggest:
+    lhs: fmap and (sequence s)
+    rhs: Universum.andM s
+    note: |
+      Applying this hint would mean that some actions
+      that were being executed previously would no longer be executed.
+- suggest:
+    lhs: and <$> sequence s
+    rhs: Universum.andM s
+    note: |
+      Applying this hint would mean that some actions
+      that were being executed previously would no longer be executed.
+- suggest:
+    lhs: fmap or (sequence s)
+    rhs: Universum.orM s
+    note: |
+      Applying this hint would mean that some actions
+      that were being executed previously would no longer be executed.
+- suggest:
+    lhs: or <$> sequence s
+    rhs: Universum.orM s
+    note: |
+      Applying this hint would mean that some actions
+      that were being executed previously would no longer be executed.
+- suggest:
+    lhs: fmap and (mapM f s)
+    rhs: Universum.allM f s
+    note: |
+      Applying this hint would mean that some actions
+      that were being executed previously would no longer be executed.
+- suggest:
+    lhs: and <$> mapM f s
+    rhs: Universum.allM f s
+    note: |
+      Applying this hint would mean that some actions
+      that were being executed previously would no longer be executed.
+- suggest:
+    lhs: fmap or (mapM f s)
+    rhs: Universum.anyM f s
+    note: |
+      Applying this hint would mean that some actions
+      that were being executed previously would no longer be executed.
+- suggest:
+    lhs: or <$> mapM f s
+    rhs: Universum.anyM f s
+    note: |
+      Applying this hint would mean that some actions
+      that were being executed previously would no longer be executed.
+
+# Unfortunately, these are often bad because they remove a variable name
+# (which usually clarifies things):
+# - suggest: { lhs: (do x <- m; when x a), rhs: Universum.whenM m a }
+# - suggest: { lhs: (do x <- m; unless x a), rhs: Universum.unlessM m a }
+
+- warn: { lhs: whenM (not <$> x), rhs: unlessM x }
+- warn: { lhs: unlessM (not <$> x), rhs: whenM x }
+
+# Oh boy, we sure have many ways of spelling “pure ()”. Also I checked and
+# HLint isn't smart enough to see reordered case branches.
+- warn: { lhs: (case m of Just x -> f x; Nothing -> pure ()  ), rhs: Universum.whenJust m f }
+- warn: { lhs: (case m of Just x -> f x; Nothing -> return ()), rhs: Universum.whenJust m f }
+- warn: { lhs: (case m of Just x -> f x; Nothing -> pass     ), rhs: Universum.whenJust m f }
+- warn: { lhs: (case m of Nothing -> pure ()  ; Just x -> f x), rhs: Universum.whenJust m f }
+- warn: { lhs: (case m of Nothing -> return (); Just x -> f x), rhs: Universum.whenJust m f }
+- warn: { lhs: (case m of Nothing -> pass     ; Just x -> f x), rhs: Universum.whenJust m f }
```

Comment:
I think that sometimes `case` version is better, mostly when `m` and `f` are big.

Reply:
I don't think that the size of `f` matters, since it is used like so:
```
whenJust m $ do
    <make your f as big as you like>
```
as for `m`, I don't see much difference between `case m of` and `whenJust m $ do`. The latter is slightly longer, but that's just a few characters.